File: Java/q1045.java
// Tipos de triângulos

import java.util.Scanner;

public class Main {
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);
        double a = input.nextDouble();
        double b = input.nextDouble();
        double c = input.nextDouble();
        double A, B, C;
        
        if (a > b){
            if (a > c){
                A = a;
                B = c;
                C = b;
            } else {
                A = c;
                B = a;
                C = b;
            }
         } else {
             if (b > c){
                 A = b;
                 B = c;
                 C = a;
            } else {
                A = c;
                B = b;
                C = a;
            }
         }
         
         if (A >= B+C){
             System.out.println("NAO FORMA TRIANGULO");
         } else if ((A*A) == (B*B)+(C*C)){
             System.out.println("TRIANGULO RETANGULO");
         } else if ((A*A) > (B*B)+(C*C)){
             System.out.println("TRIANGULO OBTUSANGULO");
         } else if ((A*A) < (B*B)+(C*C)){
             System.out.println("TRIANGULO ACUTANGULO");
         }
         
         if (A == B && B == C){
             System.out.println("TRIANGULO EQUILATERO");
         } else if (A == C || B == C || A == B){
             System.out.println("TRIANGULO ISOSCELES");
         }
    }
}

File: Cpp/q1045.cpp
#include <iostream>
#include <algorithm>

using namespace std;
 
int main(){

  float A, B, C, arr[3];
  int t;
  cin >> arr[0] >> arr[1] >> arr[2];

  t = sizeof(arr)/sizeof(arr[0]);
  sort(arr, arr+t);
  A = arr[2];
  B = arr[1];
  C = arr[0];

  if(A >= B+C){

    cout << "NAO FORMA TRIANGULO" << endl;

  } else if((A*A) == (B*B)+(C*C)){

    cout << "TRIANGULO RETANGULO" << endl;

  } else if((A*A) > (B*B)+(C*C)){

    cout << "TRIANGULO OBTUSANGULO" << endl;

  } else if((A*A) < (B*B)+(C*C)){

    cout << "TRIANGULO ACUTANGULO" << endl;

  }

  if(A == B && B == C){

    cout << "TRIANGULO EQUILATERO" << endl;

  } else if(A == C || B == C || A == B){

    cout << "TRIANGULO ISOSCELES" << endl;

  }
  
  return 0;
}